components/renku_data_services/notebooks/crs.py

```diff
@@ -32,6 +32,8 @@
 from renku_data_services.notebooks.cr_amalthea_session import Model as _ASModel
 from renku_data_services.notebooks.cr_amalthea_session import Resources3 as Resources
 from renku_data_services.notebooks.cr_amalthea_session import Secret1 as SecretAsVolume
+from renku_data_services.notebooks.cr_amalthea_session import SecretRef as SecretRefKey
+from renku_data_services.notebooks.cr_amalthea_session import SecretRef1 as SecretRefWhole
```

Comment:
question: `SecretRef1` seems like a pretty bad name in the originating file, could this be changed or is this something autogenerated that we can't do anything about?

Reply:
It is auto-generated. Everything in `cr_jupyter_server.py` and `cr_amalthea_session.py` is autogenerated. That is why I import from these two files into `crs.py` and I rename there. There is no way to change the names in the autogenerated code.

Reply:
can it be changed on the amalthea side where it comes from?

Reply:
> can it be changed on the amalthea side where it comes from?

No I mean I would have to change the CRD spec for amalthea sessions just so I have a nice name when I autogenerate code. That does not make sense. And the reason there is multiple is that the code generation does not deduplicate all things. And there are multiple `SecretRef` fields in the CRD spec. So TLDR No.

Reply:
ok, makes sense